pinot-common/src/main/java/com/linkedin/pinot/common/utils/EqualityUtils.java

```diff
@@ -40,11 +39,11 @@ public static boolean isEqual(long left, long right) {
   }
 
   public static boolean isEqual(float left, float right) {
-    return Float.compare(left, right) == 0;
+    return left == right;
   }
 
   public static boolean isEqual(double left, double right) {
-    return Double.compare(left, right) == 0;
+    return left == right;
```

Comment:
To be honest, I have the same concern. Compare to '==', the overhead of one method call is too expensive, so we should avoid using this. But there are some existing code already use these for primitive types...